producer/example_test.go

```diff
@@ -0,0 +1,61 @@
+package producer_test
+
+import (
+	"context"
+
+	"github.com/Shopify/sarama"
+	"github.com/heetch/felice/codec"
+	"github.com/heetch/felice/producer"
+)
+
+var endpoints []string
+
+func Example() {
```

Comment:
I really like the examples in the producer package, but I think it's now complex enough that we could do with a `doc.go` there too.  Just some description to lead into the examples.   We might consider renaming `example_test.go` to `doc_test.go` just to indicated the relationship too.